### PR TITLE
Adds an actual oven

### PR DESCRIPTION
**Pull request by MadmanMartian**
Adds a simple electric oven. Put reagent container in, it heats it over time while consuming power.

Upgrading its internal parts reduce the active_power_usage strain on the network, increase its maximum heat it can get to, and how quickly it can heat something.

TODO:

- [x] Add the oven to RnD
- Add the oven to roundstart kitchens - Requires 16812 to be merged, or this to be merged first and it changed
- [x] Add the oven to ghetto circuitry(?)
- [x] Figure out aggressive grab-code so you can shove people in ovens.

**Related PRs to the kitchen rework:**

- #16820 Reworks the Ice cream machine to contain and use resources
- #16812 Removes the weird and wonderful goofkitchen machines
- #16829 Adds an actual oven

:cl:
* rscadd: Adds an electric oven that can be used for baking or heating things

### CODE REVIEW

File: code/modules/food/cooking_machines.dm [hunk 1]
```diff
@@ -62,7 +62,7 @@ var/global/ingredientLimit = 10
 // Base (Oven) /////////////////////////////////////////////////
 
 /obj/machinery/cooking
-	name = "oven"
+	name = "Waffle Inc. Ez-bake FUN oven"
```
Comment (ghost) [inline code/modules/food/cooking_machines.dm:65]: unatomic

**Comment by ghost** [comment]
So this is just yet another bunsen burner?

**Comment by Unknown-as-Captain** [comment]
So this is just yet another bunsen burner?

**Comment by MadmanMartian** [comment]
well I can't add the recipes for the fun pies and funbreads yet.

It'd be unatomic.

File: code/modules/food/cooking_machines.dm (hunk 1 above)
Comment (MadmanMartian) [reply code/modules/food/cooking_machines.dm:65]: Needed to differentiate the two, else they'd share the same sprites and the same name.

**Comment by ghost** [comment]
Not really, no

**Comment by Chazach** [comment]
>can't cram people inside

oy vey

File: code/modules/food/cooking_machines.dm [hunk 2]
```diff
@@ -679,3 +679,89 @@ var/global/ingredientLimit = 10
...
+	else if(use_power == 2)
+		toggle(user)
+
+/obj/machinery/oven/proc/toggle(mob/user)
+	if(use_power == 1)
+		icon_state = icon_state_on
+		to_chat(user, "<span class = 'notice'>You turn \the [src] on.</span>")
+		use_power = 2
+		processing_objects.Add(src)
+		return
```
Comment (DamianX) [inline code/modules/food/cooking_machines.dm]: Don't need to return here nor in the other branch

**Review by DamianX — DISMISSED** [review]
You should check for power inside `process` or it will work without it.
I think this needs `density = TRUE` and `anchored = TRUE` unless we want people to walk over the oven and drag it around.
It needs `machine_flags` to allow deconstruction/unanchoring/upgrading.

**Comment by SGuivenne** [comment]
yeah, we gotta take this to the obvious next step of shoving things or people into the oven.  Or leaving the door open producing carbon dioxide

**Comment by Fruktlimpa** [comment]
I agree, make it possible to toss someone inside via aggressive grab.
Allow the person inside to resist out after some time

**Comment by SkowronX** [comment]
Watch out, Shlomo.

**Comment by Hugh-G-Rection** [comment]
I like where this is going
image

**Comment by SGuivenne** [comment]
Also would this resolve several current oven related bugs, such as #16590, and #16730 ?

**Comment by MadmanMartian** [comment]
the prior possibly, the latter would mean some cooking datums specifically for the oven would need to be made.

I'm going to pick this back up after antag datums, so I will complete the latter

**Comment by DamianX** [comment]
Conflicts, inactive for a month, he's busy with antag datums
Feel free to reopen

**Comment by jknpj** [comment]
ship it

**Pull request merged**